server/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -379,6 +385,37 @@ public static GetResult readGetResult(StreamInput in) throws IOException {
         return result;
     }
 
+    private Map<String, DocumentField> readFields(StreamInput in) throws IOException {
+        Map<String, DocumentField> fields = null;
+        int size = in.readVInt();
+        if (size == 0) {
+            fields = new HashMap<>();
+        } else {
+            fields = new HashMap<>(size);
+            for (int i = 0; i < size; i++) {
+                DocumentField field = DocumentField.readDocumentField(in);
+                fields.put(field.getName(), field);
+            }
+        }
+        return fields;
+    }
+    
+    public static void splitFieldsByMetadata(Map<String, DocumentField> fields, Map<String, DocumentField> outOther,
```

Comment:
should be drop `public` modifier here?

Reply:
Unfortunately this method is required in tests e.g. https://github.com/elastic/elasticsearch/pull/38373/files#diff-219c73a7a0b84de3e96bdda87ec92270R218 when we are building random getResult using the builders of random DocumentFields and need to understand where to put each of the randomly generated fields. Not sure how to go around it.

Reply:
@sandmannn removing `public` modifier will make this method package-private which will still allow it to be used in `GetResultTests` class.

Reply:
thanks, done